### ISSUE TITLE
Critical bug in MetaModel::save

### ISSUE THREAD

**Issue by dmolineus** [MetaModels/core#894]
When a MetaModel item which is loaded with a subset of it's attributes is saved using the API all other attributes values will get lost when saving the item.

``` php
<?php

$item     = $metaModel->findById(5, array('visitors'));
$visitors = $item->get('visitors') + 1;

$item->set('visitors', $visitors);
$item->save(); // Now all other attributes when visitors get lost
```

**Comment by discordier** [comment]
Since when? Ususally the item should only loop over the attributes contained in the item.

**Comment by dmolineus** [comment]
Don't know since when. But it looks like it loops over all attributes:
https://github.com/MetaModels/core/blob/master/src/MetaModels/MetaModel.php#L1006-L1008

**Comment by discordier** [comment]
Yes it is... will get resolved when rewriting the saving for variants and copying of these. @stefanheimes please keep this in mind.

**Comment by stefanheimes** [comment]
@dmolineus Hi can you check the Hotfix. c05a125ddaefbd9463b0479c2e42165cbecbebf2 If you say everything is okay i will merge it into the maser and dev.

**Comment by stefanheimes** [comment]
The bugfix was added with the 2.0.0-alpha3. Pleas test it with this version or the current one 2.0.0-alpha6. @dmolineus

**Comment by MiniModel** [comment]
This issue has been labeled with "testing" but got no feedback.

We are reliant on feedback to prove a ticket fixed as we can not tell if the ticket has been
resolved if we receive no feedback.
I will close this ticket now, as I assume it has been resolved.
If the ticket should still be valid, please feel free to reopen it.

@dmolineus could you please have a final look at this and tell if your issue has been resolved?

**Comment by richardhj** [comment]
I want to revert 588eddf8a619f876d62fbb009f389666ec46bfd2.
The only attributes that implement the `modelSaved()` are the alias attribute and the combinedvalues attribute. That attributes surely don't cause data loss.
I doubt that this hotfix fixed the problem.
On the counter side I more and more encounter problems with the changes (combinedvalues do not get processed).

**Comment by discordier** [comment]
Well, the original issue (overwriting non contained attributes with `null`) has been resolved.

The problem with combinedvalues/alias is:
1. they need all source attribute data to be present in the `Item`.
2. they must be processed after their source attributes (consider a configuration: `alias` => `combined-values` => `alias`)

These must get adressed but so far no one has tackled this problem and therefore no one had a good idea how to solve this. The `modelSaved()` method was a partial solution to the problem (as it tried to address point 2. above).

**Comment by zonky2** [comment]
> Well, the original issue (overwriting non contained attributes with null) has been resolved.

@richardhj we can close this issue? may be we open another issue for a new configuration with alias => combined-values => alias... (MM 2.2)
